sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4

```diff
@@ -474,6 +475,14 @@ selectClause
     : SELECT (hints+=hint)* setQuantifier? namedExpressionSeq
     ;
 
+setClause
+    : SET assign (',' assign)*
+    ;
+
+assign
+    : key=multipartIdentifier EQ value=valueExpression
```

Comment:
valueExpression -> expression. 

Although we do not support subqueries, we should make the parser support it and issue an exception in the analyzer.

Reply:
Updated.